exchanges/gateio/ws_ob_update_manager.go

```diff
@@ -99,10 +100,26 @@ func (m *wsOBUpdateManager) LoadCache(p currency.Pair, a asset.Item) *updateCach
 
 // SyncOrderbook fetches and synchronises an orderbook snapshot to the limit size so that pending updates can be
 // applied to the orderbook.
-func (c *updateCache) SyncOrderbook(ctx context.Context, g *Gateio, pair currency.Pair, a asset.Item, limit uint64) error {
-	// TODO: When templates are introduced for all assets define channel key and use g.Websocket.GetSubscription(ChannelKey{&subscription.Subscription{Channel: channelName}})
-	// to get the subscription and levels for limit. So that this can scale to config changes. Spot is currently the only
-	// asset with templates but it has only one level.
+func (c *updateCache) SyncOrderbook(ctx context.Context, g *Gateio, pair currency.Pair, a asset.Item) error {
+	// TODO: When templates are introduced for all assets, scale to dynamic supported limit/levels provided from config.
+	var limit uint64
+	switch a {
+	case asset.Spot:
+		sub := g.Websocket.GetSubscription(spotOrderbookUpdateKey)
```

Comment:
This is super cute. Never thought of it. Really like the pattern.

Reply:
❤️